src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -880,40 +898,52 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			connection.options['azureAccountToken'] = tokenPromise.token;
 			return true;
 		}
+		let account = this.findAzureAccountFromProfile(accounts, connection);
+		if (account) {
+			this._logService.debug(`Getting security token for Azure account ${account.key.accountId}`);
+			if (account.isStale) {
+				this._logService.debug(`Account is stale - refreshing`);
+				try {
+					account = await this._accountManagementService.refreshAccount(account);
+				} catch (err) {
+					this._logService.info(`Exception refreshing stale account : ${toErrorMessage(err, true)}`);
+					// refreshAccount throws an error if the user cancels the dialog
```

Comment:
This comment doesn't really explain much here - especially with why returning false is appropriate here. Would suggest making it clearer how the comment and the return value are related.